### PR TITLE
vmware_vm_shell: speed up vm_name search

### PR DESCRIPTION
**Pull request by ritzk**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

ansible-modules-extras/cloud/vmware/vmware_vm_shell.py
##### ANSIBLE VERSION

```
ansible 2.0.1.0
```
##### SUMMARY

blocked_by: https://github.com/ansible/ansible/pull/15053
- vm_name can be shared across multiple datacenter. Eliminate this mismatch.
- Makes search based on vm_name faster, as we limit search to a given datacenter.

```
Without the fix
[ritesh@fedorabox ansible (vmware_fixes)]$ time ./hacking/test-module  -m ~/Projects/github/ritz/ansible-modules-extras/cloud/vmware/vmware_vm_shell.py   -a "hostname=myESXi datacenter=myDC validate_certs=false  username=myUserName password=myPassword vm_username=groot vm_password=vmPassword vm_shell=/bin/echo  vm_shell_args=Hello vm_id=myVM " 
...
real    33m5.086s
user    0m31.141s
sys     0m3.762s

With the fix
real    0m24.749s
user    0m0.876s
sys     0m0.164s

```

### CODE REVIEW

**Comment by resmo** [comment]
Thanks @ritzk. Since you are one of the maintainers (@ritzk, @chrrrles) of this module, we are marking this PR for inclusion.

**Comment by ritzk** [comment]
@resmo , this is blocked by https://github.com/ansible/ansible/pull/15053 . Once this has been merged, we can push the patch in.

**Comment by gregdek** [comment]
Thanks again to @ritzk. This module is going into community review.

We notice that you are one of the maintainers (@ritzk, @chrrrles, @jcpowermac, @mtnbikenc) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by resmo** [comment]
since https://github.com/ansible/ansible/pull/15053 has been merge, and this looks like a reasonable improvement, could you (@ritzk, @chrrrles, @jcpowermac, @mtnbikenc) review asap and mark it as shipit so we get this into 2.1 before the feature freeze (today or tomorrow!)

File: cloud/vmware/vmware_vm_shell.py [hunk 1]
```diff
@@ -156,18 +141,33 @@ def main():
...
+        vm = find_vm_by_id(content, p['vm_id'], p['vm_id_type'], datacenter, cluster)
         if not vm:
-            module.fail_json(msg='failed to find VM')
+            module.fail_json(msg='VM not found')
 
         msg = execute_command(content, vm, p['vm_username'], p['vm_password'], 
                               p['vm_shell'], p['vm_shell_args'], p['vm_shell_env'], p['vm_shell_cwd'])
 
-        module.exit_json(changed=False, virtual_machines=vm.name, msg=msg)
+        module.exit_json(changed=False, uuid=vm.summary.config.uuid, msg=msg)
```
Comment (chrrrles) [inline cloud/vmware/vmware_vm_shell.py]: Hi @ritzk - I overlooked this in the original PR...  Did we want to exit here with changed=False?

Comment (ritzk) [inline cloud/vmware/vmware_vm_shell.py]: > changed=False
> I would prefer to say True.
> 
> vm.name
> uuid is usually a lot more unique.

**Comment by chrrrles** [comment]
+1

**Comment by gregdek** [comment]
Thanks again to @ritzk for this PR, and thanks @ritzk, @chrrrles, @jcpowermac, @mtnbikenc for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**